doc/how-to/how-to-grant-metabase-permissions.md

```diff
@@ -0,0 +1,51 @@
+# How to grant Metabase permissions
+
+## What is Metabase?
+[Metabase](https://www.metabase.com/) is an open source BI service which we self-host as part of PlanX. It allows teams to view and self-serve analytics dashboards related to their flows, applications, and users.
+
+Metabase is set up and running on both Staging and Production environments, but only the Production instance (with Production data) has dashboards maintained and curated for teams.
+
+## Context
+Metabase accesses our staging and production databases through the `metabase_read_only` role which has `SELECT` (read-only) access to a subset of tables and views. This ensures that sensitive user data cannot be inadvertently exposed via Metabase, and any new tables added to the PlanX database have to be explicitly exposed via this role.
+
+The permissions granted for the `metabase_read_only` role are applied via Hasura migrations. This ensures that we have a version-controlled history of this role, and it's access level is documented in code. This is not controlled via Pulumi (IaC) as this is not used for local development or test environments ("Pizzas"), which would necessitate a second method for these environments.
+
+## How is this role used?
+
+### Staging & Production
+The `metabase_read_only` role is granted to the `metabase_user` database user, which is manually set up on both staging and production with the following SQL -
+
+```sql
+CREATE USER metabase_user WITH PASSWORD <$PASSWORD>;
+GRANT metabase_read_only TO metabase_user;
+```
+
+The password for Staging and Production databases can be found the OSL 1Password account.
```

Comment:
This is the solution I'm proposing for setting up the `metabase_user` user. Not perfect but hopefully workable?

Chances are it will exist on staging and production and we'll never really need to deal with it locally.

Reply:
This seems like a good thoughtful approach and will make for less-hassle rotation if necesary in future, but not entirely following based on this doc how this user relates to existing `metabase` user role provisioned/managed by Pulumi - eg https://github.com/theopensystemslab/planx-new/blob/main/infrastructure/application/index.ts#L100 

At minimum, do we need to think about more distinctive naming? At most, is there a way to consolidate responsibilities of each into a single user role?

Reply:
Yeah this caught me out initially also.

This section of code is generating a role and user to be used for the `MB_DB_CONNECTION_URI` env var, which is responsible for the schema and role of the internal metabase db, where dashboards etc are stored - not the PlanX schema where data is accessed via Hasura / ShareDB.

<img width="287" alt="image" src="https://github.com/theopensystemslab/planx-new/assets/20502206/8e720172-b0f3-439f-9235-dda701002f4d">

I'll add a code comment to Pulumi to make this explicit, as well as document this in the "how to" in this PR.

Do you think this would be enough?

Reply:
Yep happy with extra docs here !